### ISSUE TITLE
Infinite exp(?)

### ISSUE THREAD

**Issue by Laifsyn** [eideehi/mc-experiencebottler#12]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Mod version

1.19.4+build.1

### Is this a bug caused by this mod alone?

The bug was reproduced with this mod alone

### What happened?

> Set to level 18. Check and Screenshot xp Bottler
![image](https://github.com/eideehi/mc-experiencebottler/assets/99366187/6935329c-7b8b-4741-9577-4bd296f7a797)
> Add 1 exp point through command . Check and screenshot xp Bottler
![image](https://github.com/eideehi/mc-experiencebottler/assets/99366187/7165dc1d-73f0-454c-9aa3-9ea7429d9be6)

### Steps to reproduce the bug

Gonna check if it's reproducible in vanilla tomorrow

**Comment by Laifsyn** [comment]
Hey. Follow up.
https://youtu.be/BZ1hB2G9e6Q

mods used at the test:
```

PuzzlesLib-v6.0.11-1.19.4-Fabric.jar
cloth-config-10.0.96-fabric.jar
EasyAnvils-v6.0.0-1.19.4-Fabric.jar
experiencebottler-1.19.4+build.1.jar
fabric-api-0.83.0+1.19.4.jar
fabric-language-kotlin-1.9.5+kotlin.1.8.22.jar
ForgeConfigAPIPort-v6.0.2-1.19.4-Fabric.jar
health-levels-1.0.5.jar
```

**Comment by eideehi** [comment]
It seems to be a bug that occurred from commit ed54a30eaf6cd94a5260969d3308b11acf01c31c. It has been fixed in v1.20.1+build.1. Please note that there is no release for v1.19.4+build.2 due to the older Minecraft version.

**Comment by eideehi** [comment]
From now on, please try to describe the issue as much as possible in text rather than relying solely on screenshots or videos. While screenshots and videos are valuable sources of information, they should primarily complement the written description.

**Comment by Laifsyn** [comment]
well. In this case it was hard for me because to properly give a description, I would've to be able to properly understand what was happening. I wasn't expecting a bug from a improved calculation logic. 
So the 201a91774d132a2d0873d715f4ef6d1d78cb9aca commit should be safe to use? how would I go about to compile it as a jar for my server?

**Comment by Laifsyn** [comment]
nvm. I'll just cope with it until I can upgrade to 1.20

**Comment by eideehi** [comment]
To build the latest patched version 1.19.4, follow these steps (git needs to be installed):

1. Open the terminal in any directory.
2. Execute the command `git clone https://github.com/eideehi/mc-experiencebottler.git`
3. Execute the command `cd mc-experiencebottler`
4. Execute the command `git checkout 1.19.4+build.1`
5. Edit the file `mc-experiencebottler/src/main/java/net/eidee/minecraft/experiencebottler/util/ExperienceUtil.java` with any text editor.
    * Change line 71 to `int level = 512;`
    * Change line 110 to `return TOTAL_EXP_CACHE[index] + Math.round(getNextLevelExperience(index) * progress);`
6. Execute the command `gradlew build`

Once the build is complete, the modified `experiencebottler-1.19.4+build.1.jar` will be generated in the `mc-experiencebottler/build/libs` directory.

**Comment by Laifsyn** [comment]
working *thumbs-up
![image](https://github.com/eideehi/mc-experiencebottler/assets/99366187/914e4f95-2d96-4511-b8fa-b01992e2dfdf)
